Fix PDO pgsql for macOS with Homebrew and disable PCRE JIT for M1

hlubek opened the pull request:
* Sets the `libpq` path configure option for macOS for PDO pgsql
* Disables PCRE JIT for arm64 (e.g. Apple M1) because it doesn't work (yet)

With these fixes we were able to install PHP 7.4, 8.0 and 8.1 on macOS with Intel (older 10.15) and Apple Silicon (12.2), so it should work for a wider variety of OS versions.

Drarok commented:
This is just what I needed to get a working build of PHP working via asdf, thanks!

iseth commented:
@smorimoto any chance this could be merged?

smorimoto commented:
@hlubek Could you rebase this PR to base branch?

hlubek commented:
@smorimoto I'll rebase the change and have a look at it 👍

hlubek commented:
@smorimoto It should work now, I noticed that the logic for `PHP_WITHOUT_PCRE_JIT` was switched around somehow in 4d6388b376bfed2c4996b003ec8f38fcfd5312ff . I fixed that and added an automatic detection for Arm-based Macs (M1, M2, ...) that have issues with PCRE JIT.

lucaspbastos commented:
Any update on when we can see this merged? @smorimoto

albuquerque53 reviewed (approved):
Very nice, congratulations for the fix!
